S4TF_Tutorial_6.ipynb

```diff
@@ -0,0 +1,1356 @@
+        "    }\n",
+        "    \n",
+        "    if random.random() < 0.5 {\n",
+        "        let h = Float(img.size[0])! / 10\n",
+        "        let w = Float(img.size[1])! / 10\n",
+        "        let borderSize = Int(Python.min(h, w))\n",
+        "        img = pilImageOps.expand(img, borderSize) // Add border.\n",
+        "    }\n",
+        "    \n",
+        "    var image = np.array(img, dtype: np.float32) * (1.0 / 255)\n",
+        "    \n",
+        "    if random.random() < 0.5 {\n",
+        "        image = sk.util.random_noise(image) // Add noise.\n",
+        "    }\n",
+        "    \n",
+        "    image = np.array(image, dtype: np.float32)\n",
+        "    \n",
+        "    let imageTensor = Tensor<Float>(numpy: image)!\n",
+        "\n",
+        "    return imageTensor\n",
+        "}"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "vl6vw_AcZd54",
+        "colab_type": "text"
+      },
+      "source": [
+        "# Data Preparation "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "IQ3OILUgZeqb",
+        "colab_type": "text"
+      },
+      "source": [
+        "Images must be formatted into appropriately pre-processed floating point tensors before being fed into the network. The steps involved in preparing these images are:\n",
+        "\n",
+        "1. Read images from the disk\n",
+        "2. Decode contents of these images and convert it into proper grid format as per their RGB content\n",
+        "3. Convert them into floating point tensors\n",
+        "4. Rescale the tensors from values between 0 and 255 to values between 0 and 1, as neural networks prefer to deal with small input values.\n",
+        "5. Apply or not apply Image Augmentation Techniques, based on type of data\n",
+        "\n",
+        "We have done this in the following code."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "0FyESvV0ZjhJ",
+        "colab_type": "text"
+      },
+      "source": [
+        "The `tensor` function takes 2 inputs are:\n",
+        "1. Image path as input\n",
+        "2. Whether or not to apply Image Augmentations\n",
+        "\n",
+        "The `tensors` function takes 3 inputs :\n",
+        "\n",
+        "1. List of paths\n",
+        "\n",
+        "2. Number of tensors to be produced in the output tensor as input\n",
+        "\n",
+        "3. Whether or not to apply Image Augmentations"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "3JihjHwPi2Ch",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "// Function `tensor(path:ImageAumentation:)` returns the resized image tensor w/ augmentation and it's corresponding label.\n",
+        "func tensor(path: String, ImageAumentation: Bool) -> (Tensor<Float>, Int32){\n",
```

Comment:
1. Add a space between `(` and `{`.
2. The function base name "tensor" suggests it returns a single `Tensor`, but it also returns an `Int32`. And the base name does not say anything informative about what this function does. Please rename the function to `resizedImage(fromPath:augmented:)`, and add tuple element labels to the return type.

```suggestion
        "func resizedImage(fromPath path: String, augmented: Bool) -> (image: Tensor<Float>, label: Int32) {\n",
```